src/plugins/kibana/public/settings/sections/indices/add_data_steps/pipeline_setup/lib/processor_types.js

```diff
@@ -0,0 +1,42 @@
+export class Processor {
+  constructor(processorId) {
+    this.processorId = processorId;
+    this.typeId = 'base';
+    this.collapsed = false;
+    this.parent = undefined;
+    this.inputObject = undefined;
+    this.outputObject = undefined;
+    this.error = undefined;
+  }
+
+  setParent(newParent) {
+    const oldParent = this.parent;
+    this.parent = newParent;
+
+    return (oldParent !== this.parent);
+  }
+}
+
+export class SetProcessor extends Processor {
+  constructor(processorId) {
+    super(processorId);
+    this.title = 'Set';
+    this.typeId = 'set';
```

Comment:
Does every processor have a typeId and a title? If so, I would make them parameters on the base class's constructor and pass them in that way instead of setting them directly on `this`. You could then throw an error in the base class constructor if those params aren't included, to ensure that every subclass provides them.